src/server/scripts/World/server_mail.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "Mail.h"
+#include "ObjectMgr.h"
+#include "QueryResult.h"
+
+class ServerMailReward : public PlayerScript
+{
+public:
+    ServerMailReward() : PlayerScript("ServerMailReward") { }
+
+    // CHARACTER_LOGIN = 8
+    void OnLogin(Player* player) override
+    {
+        for (auto const& servMail : sObjectMgr->GetAllServerMailStore())
+        {
+            CharacterDatabasePreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_MAIL_SERVER_CHARACTER);
+            stmt->SetData(0, player->GetGUID().GetCounter());
+            stmt->SetData(1, servMail.second.id);
+
+            WorldSession* mySess = player->GetSession();
+            mySess->GetQueryProcessor().AddCallback(CharacterDatabase.AsyncQuery(stmt)
+                .WithPreparedCallback([mySess](PreparedQueryResult result)
+                    {
+                        for (auto const& servMail : sObjectMgr->GetAllServerMailStore())
+                        {
+                            if (result)
+                                continue;
```

Comment:
I really do not understand how this always works. But is skipped if you do .reload mail_server_template

even though all data is the same in DB?

Reply:
It only happens when you reload after adding a new entry. not everytime you reload.

Could it have something to do with the server mail store? unsure